api/agent.go

```diff
@@ -237,6 +237,56 @@ func (a *Agent) Health() (*AgentHealthResponse, error) {
 	return nil, fmt.Errorf("unable to unmarshal response with status %d: %v", resp.StatusCode, err)
 }
 
+// Monitor returns a channel which will receive streaming logs from the agent
+// Providing a non-nil stopCh can be used to close the connection and stop log streaming
+func (a *Agent) Monitor(stopCh <-chan struct{}, q *QueryOptions) (<-chan *StreamFrame, <-chan error) {
+	errCh := make(chan error, 1)
+	r, err := a.client.newRequest("GET", "/v1/agent/monitor")
+	if err != nil {
+		errCh <- err
+		return nil, errCh
+	}
+
+	r.setQueryOptions(q)
+	_, resp, err := requireOK(a.client.doRequest(r))
+	if err != nil {
+		errCh <- err
+		return nil, errCh
+	}
+
+	frames := make(chan *StreamFrame, 10)
+	go func() {
+		defer resp.Body.Close()
+
+		dec := json.NewDecoder(resp.Body)
+
+		for {
+			select {
+			case <-stopCh:
```

Comment:
is it worthwhile to close `frames` here as well? the code seems fine either way, because the FrameReader gets `stopCh` as its cancel-channel, used in the same `select` as `frames`, alongside the `errCh`. however, it does follow the same pattern that we use below:
https://github.com/hashicorp/nomad/pull/6499/files?utf8=%E2%9C%93&diff=unified&w=1#diff-4f360b6361d9900f98e2f244a3411bc4R273-R274

Reply:
Yeah I think we should close here as well to be consistent